packages/CodeGenLib/src/Database/manage-metadata.ts

```diff
@@ -25,6 +31,93 @@ export class ManageMetadataBase {
       return this._newEntityList;
    }
 
+   public async enableMetaDataLogging(config: MetadataLoggingConfig): Promise<void> {
+
+      if(!config){
+         logError("MetadataLoggingConfig is required to enable metadata logging");
+         return;
+      }
+
+      if(!config.folderPath && !config.filePath){
+         logError("Either folderPath or filePath is required to enable metadata logging");
+         return;
+      }
+
+      if(config.folderPath){
+
+         const dirExists: boolean = fs.existsSync(config.folderPath);
+         if(!dirExists){
+            fs.mkdirSync(config.folderPath, {recursive: true });
+         }
+
+         const fileName: string = this.createFileName();
+         this.FilePath = path.join(config.folderPath,`${fileName}.sql`);
+
+         //create an empty file
+         fs.writeFileSync(this.FilePath, '');
+      }
+      else{
+         this.FilePath = config.filePath;
+         if(this.FilePath && config.overwriteFile){
+            //append to the file
+            fs.writeFileSync(this.FilePath, '');
+         }
+      }
+
+      this.Config = config;
+      LogStatus(`Metadata logging enabled. File path: ${this.FilePath}`);
+   }
+
+   protected createFileName(): string {
+      const date = new Date();
+
+      const year = date.getUTCFullYear();
+      const month = String(date.getUTCMonth() + 1).padStart(2, '0'); // Month is 0-based
+      const day = String(date.getUTCDate()).padStart(2, '0');
+
+      const hour = String(date.getUTCHours()).padStart(2, '0');
+      const minute = String(date.getUTCMinutes()).padStart(2, '0');
+      const second = String(date.getUTCSeconds()).padStart(2, '0');
+            
+      const fileName = `CodeGen_Run_${year}-${month}-${day}_${hour}-${minute}-${second}`;
+      return fileName;
+   }
+
+   protected async appendToFile(contents: string, description?: string): Promise<void> {
+      try{
+         if(!contents || !this.FilePath){
+            return;
+         }
+
+         if(description){
+            const comment = `/* ${description} */\n`;
+            contents = `${comment}${contents}`;
+         }
+
+         contents = `${contents}\n\n`;
+
+         fs.appendFileSync(this.FilePath, contents);
+      }
+      catch(ex){
+         logError("Unable to log metadata SQL text to file", ex);
+      }
+   }
+
+   protected convertToMigrationFile(): void {
```

Comment:
@JS-BC now I see what you mean, but your comment in the JSDoc above should have been more descripve